bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/model/Partition.java

```diff
@@ -34,9 +34,11 @@ public class Partition extends Entity implements Commandable {
     private final Logger logger = LoggerFactory.getLogger(Partition.class);
 
     private PartitionState state = new PartitionState();
+    private IParadoxCommunicator communicator;
 
-    public Partition(int id, String label) {
+    public Partition(IParadoxCommunicator com, int id, String label) {
```

Comment:
Adding communicator to the model object partition loads unnecessary the Partition class which is meant to work as a model bean storing data from the caches.
It was not very well designed from the beginning by me and relied on the static retrieval of the single panel before.
My proposal is to remove completely the Commandable interface and move the handleCommand logic to the ParadoxPartitionHandler where you only need to retrieve the partition ID from the model object and pass it to the communicator command.
What do you think?

Reply:
Actually, I think that partition should have it's Panel reference, Panel has the Communicator and the command should be sent to it's communicator in handleCommand(). I'm not seeing why _adding communicator to the model object partition loads unnecessary the Partition class_. Can you please what you mean by loads the Partition class?! It is only a pointer to communicator right now but I think it should be to actually to panel.
We could change also the interface to accept a ParadoxPanel or Communicator parameter.

Reply:
In my opinion this is unnecessary dependency to the communicator / panel. The hierarchy should be Panel which consists of partitions/zones/other entities (doors?). I don't even remember why I did it like this... probably had an idea to command also doors (not supported) / zones (nothing to control there AFAIK) and every object to know by itself what commands to send to the panel/communicator. Currently I see it as additional dependency from the model (data) to the communicator (the action). The only benefit I see from this now (few years after the initial implementation) is that you can use this from he Main() class as a stand alone application for testing and maybe if someone adds future support for other entities, so you can just imlement internally the handleCommand method for each entity (door for example). What benefits do you see of this implementation instead of moving it in the handler logic?

Reply:
So, would you agree to pass a refference of Panel to Entities? I think is the best approach in a object oriented programming like Java.

Reply:
Yes. I have approved the PR.

Reply:
I saw that but there are two more things to add before this is ready to be Merged:
1. Panel disposal on ParadoxIP150BridgeHandler.dispose() - without this, Bridge won't came up online after disable / enable.
2. Every Entity should have access to it's Panel and Communicator via Panel so the constructor of Entity I propose to be modified to something like:
```
public Entity(ParadoxPanel panel, int id, String label) {
    this.panel = panel;
    this.id = id;
    this.label = label.trim();
    logger.debug("Creating entity with label [{}] and ID [{}] for panel [{}]", label, id, panel);
}
```
Do you agree with this proposal?

Reply:
Yep. I saw the panel dispose and agree with that. Also if you see fit the panel reference in the sub-entity, let's leave it as proposed in the constructor. Looks OK to me.

Reply:
I'm just not sure if my review is enough for merge. Before it was expected that someone from reviewers has to approve it.